scripts/input/climate_assessment_run.R

```diff
@@ -0,0 +1,355 @@
+    date(), " climate_assessment_prepare.R: ", if (createdOutputCsv) "Created" else "Replaced", 
+    " climateAssessmentEmi '", climateAssessmentEmi, "'\n"
+)
+cat(logMsg)
+capture.output(cat(logMsg), file = logFile, append = TRUE)
+
+#
+# RUN CLIMATE ASSESSMENT
+#
+cfg <- read_yaml(cfgPath)
+# Set default values for the climate assessment config data in case they are not available for backward compatibility
+if (is.null(cfg$climate_assessment_root)) cfg$climate_assessment_root <- "/p/projects/rd3mod/python/climate-assessment/src/"
+if (is.null(cfg$climate_assessment_infiller_db)) cfg$climate_assessment_infiller_db <- "/p/projects/rd3mod/climate-assessment-files/1652361598937-ar6_emissions_vetted_infillerdatabase_10.5281-zenodo.6390768.csv"
+if (is.null(cfg$climate_assessment_magicc_bin)) cfg$climate_assessment_magicc_bin <- "/p/projects/rd3mod/climate-assessment-files/magicc-v7.5.3/bin/magicc"
+if (is.null(cfg$climate_assessment_magicc_prob_file_iteration)) cfg$climate_assessment_magicc_prob_file_iteration <- "/p/projects/rd3mod/climate-assessment-files/parsets/RCP20_50.json"
+if (is.null(cfg$climate_assessment_r_gams_dir)) cfg$climate_assessment_r_gams_dir <- "/p/system/packages/gams/43.4.1"
+
+# The base name, that climate-assessment uses to derive it's output names
+baseFn <- sub("\\.csv$", "", basename(climateAssessmentEmi))
+
+# Auxiliary input data for climate-assessment and MAGICC7
+infillingDatabaseFile <- normalizePath(cfg$climate_assessment_infiller_db, mustWork = TRUE)
+probabilisticFile <- normalizePath(cfg$climate_assessment_magicc_prob_file_iteration, mustWork = TRUE)
+
+# Extract the location of the climate-assessment scripts and the MAGICC binary from cfg.txt
+scriptsDir <- normalizePath(file.path(cfg$climate_assessment_root, "scripts"))
+magiccBinFile <- normalizePath(file.path(cfg$climate_assessment_magicc_bin))
+magiccWorkersDir <- file.path(normalizePath(climateTempDir), "workers")
+gamsRDir <- normalizePath(cfg$climate_assessment_r_gams_dir)
+
+# Read parameter sets file to ascertain how many parsets there are
+allparsets <- read_yaml(probabilisticFile)
+nparsets <- length(allparsets$configurations)
+
+logMsg <- paste0(date(), " =================== SET UP climate-assessment scripts environment ===================\n")
+cat(logMsg)
+capture.output(cat(logMsg), file = logFile, append = TRUE)
+
+# Create working folder for climate-assessment files
+dir.create(magiccWorkersDir, recursive = TRUE, showWarnings = FALSE)
+
+#
+# SET UP MAGICC ENVIRONMENT VARIABLES
+#
+
+# Character vector of all required MAGICC7 environment variables
+magiccEnvs <- c(
+    "MAGICC_EXECUTABLE_7"    = magiccBinFile,    # Specifies the path to the MAGICC executable
+    "MAGICC_WORKER_ROOT_DIR" = magiccWorkersDir, # Directory of magicc workers
+    "MAGICC_WORKER_NUMBER"   = 1                 # TODO: Get this from slurm or nproc
+)
+
+gamsEnvs <- c(
+    "R_GAMS_SYSDIR" = gamsRDir
+)
```

Comment:
If this is for the benefit of `gdxrrw`, it is quite pointless.
1. This is run inside a REMIND run, which will not even start unless `gams` is on the PATH, where `gdxrrw` will pick it up.
2. `gdxrrw` will actually be initialised when the `remind2` package is loaded, because it has `gdx` as a dependency, which will do that explicitly.  So for `R_GAMS_SYSDIR` to have any influence on `gdxrrw` behaviour, this would have to be done before `require(remind2)`.